examples/blog/blog/build-your-own-github-using-deno-fresh-and-github-rest-api-32h6.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Build your own github using Deno Fresh and Github REST API
+description : In this post we will build a simple github using fresh and github rest api.
+slug: build-your-own-github-using-deno-fresh-and-github-rest-api-32h6
+poster : https://res.cloudinary.com/practicaldev/image/fetch/s--E2dVmnIB--/c_imagga_scale,f_auto,fl_progressive,h_420,q_auto,w_1000/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/lwrjlidkua2wn0ggd2l8.png
+
+postedAt : 19 Jul, 2022
+tags: 
+  - javascript
+  - deno
+  - fresh
+  - twind
+  - github-rest-api
+  - deno-deploy
+
+
+authorName: Harsh Mangalam
+authorAvatar: https://avatars.githubusercontent.com/u/57381638?v=4
+
+github : harshmangalam/freshHub
+
+---
+
+
+Javascript ecosystem is expanding every day. Time to time javascript enthusiastic introduce new concept and new way to enhance the developer experience like Island architecture, Resumability, fine grained reactivity and much more..
+
+In this blog post we will explore Fresh framework and will build a small github using github rest api.
+
+Deno is a simple, modern and secure runtime for JavaScript, TypeScript, and WebAssembly that uses V8 and is built in Rust.
+
+Preact is a javascript based frontend framework alternative to react with the same modern api like hooks,context,etc..
+
+
+Fresh is a server render web framework build on top of deno , preact and Island architecture pattern.
+A beautiful fact about Fresh is that it has no build so if you will deploy Fresh on Deno-Deploy it will take very less time to go live. 
+
+In this project we will use twind which provide tailwind like api for design the ui.
+
+Create project using Fresh and run locally
+
+```
+deno run -A -r https://fresh.deno.dev fresh-hub
+cd fresh-hub
+deno task start
+
+```
+
+Go inside `routes/index.tsx`
+
+```tsx
+
+/** @jsx h */
+import { h } from "preact";
+import { tw } from "@twind";
+import { PageProps, Handlers } from "$fresh/server.ts";
+import { fetchUserInfo } from "../services/github.ts";
+import Layout from "../components/Layout.tsx";
+
+export const handler: Handlers = {
+  async POST(req, ctx) {
+    try {
+      // get form data from request
+      const formData = await req.formData();
+      // extract username from form data
+      const username = formData.get("username");
+      // add validation
+      if (!username || username.toString().trim().length === 0) {
+        return ctx.render({ error: "Username should not be empty" });
+      }
+
+      // check if username exists
+      const [status] = await fetchUserInfo(String(username));
+      // handle different error status code
+      if (status === 404) {
```

Comment:
As per this PR: https://github.com/denoland/fresh/pull/417 you could change these values [(magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) to use the standard HTTP status codes: https://doc.deno.land/https://deno.land/std@0.146.0/http/http_status.ts/~/Status

Reply:
Thanks  @thegaryroberts , I added standard HTTP status codes.